sdk/compiler/daml-lf-proto-encode/test/DA/Daml/LF/Proto3/EncodeTest.hs

```diff
@@ -0,0 +1,101 @@
+-- Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Daml.LF.Proto3.EncodeTest (
+        module DA.Daml.LF.Proto3.EncodeTest
+) where
+
+
+import           Control.Monad.State.Strict
+import           Data.Int
+import qualified Data.Vector                              as V
+
+import           DA.Daml.LF.Proto3.EncodeV2
+import           DA.Daml.LF.Proto3.InternedMap
+
+import           DA.Daml.LF.Ast
+import qualified Com.Digitalasset.Daml.Lf.Archive.DamlLf2 as P
+
+import           Test.Tasty.HUnit
+import           Test.Tasty
+
+entry :: IO ()
+entry = defaultMain $ testGroup "All tests" [enc_tests]
+
+------------------------------------------------------------------------
+-- Params
+------------------------------------------------------------------------
+testVersion :: Version
+testVersion = Version V2 PointDev
+
+------------------------------------------------------------------------
+-- EncodeTests
+------------------------------------------------------------------------
+encodeKindAssert :: Kind -> P.Kind -> Assertion
+encodeKindAssert k pk =
+  let (pk', _) = runState (encodeKind k) env
+  in  pk @=? pk'
+    where
+      env = initEncodeEnv testVersion
+
+encodeKindTest :: String -> Kind -> P.Kind -> TestTree
+encodeKindTest str k pk = testCase str $ encodeKindAssert k pk
+
+enc_tests :: TestTree
+enc_tests = testGroup "Encoding tests"
+  [ enc_pure_tests
+  , enc_interning_tests
+  ]
+
+enc_pure_tests :: TestTree
+enc_pure_tests = testGroup "Encoding tests (non-interning)" $
+  map (uncurry3 encodeKindTest)
+    [ ("Kind star", KStar, pkstar)
+    , ("Kind Nat", KNat, pknat)
+    ]
+  where
+    uncurry3 :: (a -> b -> c -> d) -> ((a, b, c) -> d)
+    uncurry3 f (a, b, c) = f a b c
+
+
+enc_interning_tests :: TestTree
+enc_interning_tests = testGroup "Encoding tests (interning)"
+  [ enc_interning_starToStar
+  , enc_interning_starToNatToStar
```

Comment:
Similarly to my comment on DecodeTest, it would be useful to test that the interning logic works for Exprs that contain Kinds

Reply:
We agreed that we can defer the expression testing to the expression interning PR. Roger will leave a TODO/comment
```
-- TODO[<usernameOrInitials>]: ...
```